src/java/org/apache/cassandra/schema/MemtableParams.java

```
@@ -0,0 +1,199 @@
+        this.factory = factory;
+    }
+
+    public String configurationKey()
+    {
+        return configurationKey;
+    }
+
+    public Memtable.Factory factory()
+    {
+        return factory;
+    }
+
+    @Override
+    public String toString()
+    {
+        return configurationKey;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+
+        if (!(o instanceof MemtableParams))
+            return false;
+
+        MemtableParams c = (MemtableParams) o;
+
+        return Objects.equal(configurationKey, c.configurationKey);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return configurationKey.hashCode();
+    }
+
+    private static final String CLASS_OPTION = "class";
+    private static final String EXTENDS_OPTION = "extends";
+    private static final String DEFAULT_CONFIGURATION_KEY = "default";
+    private static final Memtable.Factory DEFAULT_MEMTABLE_FACTORY = SkipListMemtableFactory.INSTANCE;
+    private static final Map<String, String> DEFAULT_CONFIGURATION = SkipListMemtableFactory.CONFIGURATION;
+    private static final Map<String, Map<String, String>> CONFIGURATION_DEFINITIONS;
+    private static final Map<String, MemtableParams> CONFIGURATIONS;
+    public static final MemtableParams DEFAULT;
+
+    static {
+        CONFIGURATION_DEFINITIONS = expandDefinitions(DatabaseDescriptor.getMemtableConfigurations());
+        CONFIGURATIONS = new HashMap<>();
+        DEFAULT = get(null);
+    }
+
+    public static MemtableParams get(String key)
+    {
+        if (key == null)
+            key = DEFAULT_CONFIGURATION_KEY;
+
+        synchronized (CONFIGURATIONS)
+        {
+            return CONFIGURATIONS.computeIfAbsent(key, MemtableParams::parseConfiguration);
+        }
+    }
+
+    private static Map<String, Map<String, String>> expandDefinitions(Map<String, Map<String, String>> memtableConfigurations)
```

There's enough logic in `expandDefinitions()` that I'm on the fence about whether it deserves unit testing, as opposed to the coverage it has now w/ (for example) `AlterTest.testAlterTableWithMemtable()` in conjunction w/ the test `cassandra.yaml`. It will be difficult for there to be serious bugs here without them failing loudly, so I'm fine either way.

---

Added unit test.